src/shared/stores/data-value-store.js

```diff
@@ -0,0 +1,100 @@
+import create from 'zustand'
+import { CAN_HAVE_LIMITS_TYPES } from '../value-types.js'
+
+export const useStore = create((set, get) => ({
+    highlightedField: null,
+    dataValueSet: {
+        dataValues: {},
+        minMaxValues: [],
+    },
+    getDataValue: ({ dataElementId, categoryOptionComboId }) =>
+        get().dataValueSet.dataValues[dataElementId]?.[categoryOptionComboId],
+    getDataElement: ({ dataElementId }) =>
+        get().dataValueSet.dataValues[dataElementId],
+    getDataValues: () => get().dataValueSet?.dataValues,
+    getDataValueSet: () => get().dataValueSet,
+    setDataValueSet: (values) => set({ ...get(), dataValueSet: values ?? {} }),
+    getMinMaxValues: ({ dataElementId, categoryOptionComboId }) => {
+        return get().dataValueSet?.minMaxValues?.find(
+            (minMaxValue) =>
+                minMaxValue.categoryOptionCombo === categoryOptionComboId &&
+                minMaxValue.dataElement === dataElementId
+        )
+    },
+    getHighlightedField: () => get().highlightedField,
+    isFieldHighlighted: ({ dataElementId, categoryOptionComboId }) =>
+        get().highlightedField?.de?.id === dataElementId &&
+        get().highlightedField?.coc?.id === categoryOptionComboId,
+    setHighlightedField: (value) => {
+        set({ ...get(), highlightedField: value ?? null })
+    },
+    getHighlightedFieldData: () => {
+        const highlightedField = get().getHighlightedField()
+        const { de, coc } = highlightedField
+        if (!de || !coc) {
+            return undefined
+        }
+        const dataValue = get().getDataValue({
+            dataElementId: de.id,
+            cocId: coc.id,
+        })
+        const { optionSet, valueType, commentOptionSet } = de
+        const canHaveLimits = optionSet
+            ? false
+            : CAN_HAVE_LIMITS_TYPES.includes(valueType)
+        if (dataValue) {
+            return {
+                ...dataValue,
+                valueType,
+                canHaveLimits,
+                categoryOptionCombo: coc.id,
+                name: de.displayName,
+                code: de.code,
+                commentOptionSetId: commentOptionSet?.id,
+            }
+        }
+
+        return {
+            valueType,
+            canHaveLimits,
+            categoryOptionCombo: coc.id,
+            dataElement: de.id,
+            name: de.displayName,
+            lastUpdated: '',
+            followup: false,
+            comment: null,
+            storedBy: null,
+            code: null,
+            commentOptionSetId: commentOptionSet?.id,
+        }
+    },
+    hasComment: ({ dataElementId, categoryOptionComboId }) => {
+        const dataValue = get().getDataValue({
+            dataElementId,
+            categoryOptionComboId,
+        })
+
+        return !!dataValue?.comment
+    },
+    setComment: ({ dataElementId, categoryOptionComboId, comment }) => {
```

Comment:
I don't think we should update this store - then we don't share the object-reference anymore.
Also this is done in a mutation, and optimistically updated - so you should be able to just get this from the cache - without having to set it yourself.

Reply:
Have remove the use of `setComments`, the old functionality for this (in the provider that has been deleted on this PR) where originally updating a lookup object. checking on the at the dataValue object directly means we don't have to keep anything else up to date.